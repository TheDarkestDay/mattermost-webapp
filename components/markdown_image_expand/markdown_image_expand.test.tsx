// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {shallow} from 'enzyme';
import React from 'react';

import MarkdownImageExpand from './markdown_image_expand';

describe('components/MarkdownImageExpand', () => {
    it('should match snapshot for collapseDisplay set to true', () => {
        const wrapper = shallow(
            <MarkdownImageExpand
                alt={'Some alt text'}
                collapseDisplay={true}
            >{'An image to expand'}</MarkdownImageExpand>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot for collapseDisplay set to false', () => {
        const wrapper = shallow(
            <MarkdownImageExpand
                alt={'Some alt text'}
                collapseDisplay={false}
            >{'An image to expand'}</MarkdownImageExpand>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot for image collapsed with collapse button', () => {
        const wrapper = shallow(
            <MarkdownImageExpand
                alt={'Some alt text'}
                collapseDisplay={false}
            >{'An image to expand'}</MarkdownImageExpand>,
        );

        wrapper.find('.markdown-image-expand__collapse-button').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot for image expanded with expand button', () => {
        const wrapper = shallow(
            <MarkdownImageExpand
                alt={'Some alt text'}
                collapseDisplay={true}
            >{'An image to expand'}</MarkdownImageExpand>,
        );

        wrapper.find('.markdown-image-expand__expand-button').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });
});
