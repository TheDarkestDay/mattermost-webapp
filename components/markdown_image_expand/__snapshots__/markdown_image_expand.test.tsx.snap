// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MarkdownImageExpand should match snapshot for collapseDisplay set to false 1`] = `
<div
  className="markdown-image-expand markdown-image-expand--expanded"
>
  <CSSTransition
    classNames="markdown-image-expand__expanded-view"
    in={true}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__collapse-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-down"
        />
      </button>
      An image to expand
    </div>
  </CSSTransition>
  <CSSTransition
    classNames="markdown-image-expand__collapsed-view"
    in={false}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__expand-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-right markdown-image-expand__expand-icon"
        />
        <span
          className="markdown-image-expand__alt-text"
        >
          Some alt text
        </span>
      </button>
    </div>
  </CSSTransition>
</div>
`;

exports[`components/MarkdownImageExpand should match snapshot for collapseDisplay set to true 1`] = `
<div
  className="markdown-image-expand "
>
  <CSSTransition
    classNames="markdown-image-expand__expanded-view"
    in={false}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__collapse-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-down"
        />
      </button>
      An image to expand
    </div>
  </CSSTransition>
  <CSSTransition
    classNames="markdown-image-expand__collapsed-view"
    in={true}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__expand-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-right markdown-image-expand__expand-icon"
        />
        <span
          className="markdown-image-expand__alt-text"
        >
          Some alt text
        </span>
      </button>
    </div>
  </CSSTransition>
</div>
`;

exports[`components/MarkdownImageExpand should match snapshot for image collapsed with collapse button 1`] = `
<div
  className="markdown-image-expand "
>
  <CSSTransition
    classNames="markdown-image-expand__expanded-view"
    in={false}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__collapse-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-down"
        />
      </button>
      An image to expand
    </div>
  </CSSTransition>
  <CSSTransition
    classNames="markdown-image-expand__collapsed-view"
    in={true}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__expand-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-right markdown-image-expand__expand-icon"
        />
        <span
          className="markdown-image-expand__alt-text"
        >
          Some alt text
        </span>
      </button>
    </div>
  </CSSTransition>
</div>
`;

exports[`components/MarkdownImageExpand should match snapshot for image expanded with expand button 1`] = `
<div
  className="markdown-image-expand markdown-image-expand--expanded"
>
  <CSSTransition
    classNames="markdown-image-expand__expanded-view"
    in={true}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__collapse-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-down"
        />
      </button>
      An image to expand
    </div>
  </CSSTransition>
  <CSSTransition
    classNames="markdown-image-expand__collapsed-view"
    in={false}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <div>
      <button
        className="markdown-image-expand__expand-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="icon icon-menu-right markdown-image-expand__expand-icon"
        />
        <span
          className="markdown-image-expand__alt-text"
        >
          Some alt text
        </span>
      </button>
    </div>
  </CSSTransition>
</div>
`;
